doc/reference_3.1.md

```diff
@@ -0,0 +1,1049 @@
+KotlinTest
+==========
+
+[<img src="https://img.shields.io/maven-central/v/io.kotlintest/kotlintest-core.svg?label=latest%20release"/>](http://search.maven.org/#search|ga|1|kotlintest) [![GitHub license](https://img.shields.io/github/license/kotlintest/kotlintest.svg)]()
+
+This version of the document is aimed at version 3.1
```

Comment:
How long should we keep 3.1 reference for?
And apart from this line, is this the only one that changed?

Reply:
reference 3.1 is just a copy of the original reference.md with that line added.